builtin/common/vector.lua

```diff
@@ -18,40 +19,55 @@ function vector.equals(a, b)
 	       a.z == b.z
 end
 
+function vector.is_zero(a)
+	return a.x == 0 and
+	       a.y == 0 and
+	       a.z == 0
+end
+
 function vector.length(v)
 	return math.hypot(v.x, math.hypot(v.y, v.z))
 end
 
 function vector.normalize(v)
-	local len = vector.length(v)
-	if len == 0 then
-		return {x=0, y=0, z=0}
-	else
-		return vector.divide(v, len)
-	end
+	if vector.is_nil(v) then return v end
```

Comment:
don't put the body of if-statements on one line

Reply:
Should be `vector.is_zero()`?